serveradmin/views.py

```diff
@@ -42,3 +46,54 @@ def power(request):
             call(["sudo", "reboot"])
 
     return render(request, 'serveradmin/power.html')
+
+
+@staff_member_required
+def backup(request):
+    if request.POST:
+        if 'do_backup' in request.POST:
+            backup = Backup.create()
+            msg = _('Succesfully created backup {filename}').format(
+                filename=backup.name
+            )
+            messages.add_message(request, messages.SUCCESS, msg)
+        elif 'do_upload' in request.POST:
+            if 'upload' in request.FILES:
+                file_ = request.FILES['upload']
+                try:
+                    backup = Backup.load(file_)
+                except:
+                    messages.add_message(request, messages.ERROR,
+                                         _('Unable to load file.'))
+                else:
+                    msg = _('File {name} has been loaded.').format(
+                        name=backup.name)
+                    messages.add_message(request, messages.SUCCESS, msg)
+            else:
+                messages.add_message(request, messages.ERROR,
+                                     _('No file found to upload.'))
+        elif 'backup' in request.POST:
```

Comment:
All the `do_` named fields are the submit buttons, which are not anyway shipped with the Django forms.

I've chosen not to use Django forms because at the end the form is a lot of submit buttons (5 actually), which are not part of Django forms, one field that would require a custom field, because it needs Backup instance to create the HTML (a table) but on the form level it only deals with backup names, and a file field. So I made a little equation and found a Django form would mean more complexity and boostrap code.

Back to the initial question, is `do_create` a bit more clear? As we are always dealing with backup, I understand your point that having 'backup' in an action is a bit confusing.